base/threadingconstructs.jl

```diff
@@ -11,21 +10,21 @@ ID `1`.
 """
 threadid() = Int(ccall(:jl_threadid, Int16, ())+1)
 
+# lower bound on the largest threadid()
 """
-    Threads.nthreads([:default|:interactive]) -> Int
+    Threads.maxthreadid([:default|:interactive]) -> Int
 
-Get the number of threads (across all thread pools or within the specified
-thread pool) available to Julia. The number of threads across all thread
-pools is the inclusive upper bound on [`threadid()`](@ref).
+Get a lower bound on the number of threads (across all thread pools or within
+the specified thread pool) available to the Julia process >= [`threadid()`](@ref),
+with atomic-acquire semantics.
```

Comment:
I find it strange that acquire ordering is mentioned without noting what events have release store. Also, this is not an accurate description of the method `maxthreadid(pool::Symbol)` since it uses non-atomic load (the first load is OK):

https://github.com/JuliaLang/julia/blob/d635803c50b3ce48fd4d9d74638fdc07b1dd6dd0/base/threadingconstructs.jl#L38-L41

Also, my impression is that this ordering is required for runtime authors and not Julia users. For example, monotonic load with `max(threadid(), maxthreadid())` is sufficient for supporting DCL thread-local storage initialization pattern.

I think it'd be simpler to not document the ordering in the public API specification.